src/transformers/transformer-models.ts

```diff
@@ -0,0 +1,185 @@
+ */
+export interface PortableTextTable extends ArbitraryTypedObject {
+  _type: "table";
+  /**
+   * The number of columns the table has.
+   */
+  numColumns: number;
+  /**
+   * Array of table row objects.
+   */
+  rows: PortableTextTableRow[];
+}
+
+/**
+ * Represents a single row of cells in a portable text table.
+ */
+export interface PortableTextTableRow extends ArbitraryTypedObject {
+  _type: "row";
+  /**
+   * Array of table cells.
+   */
+  cells: PortableTextTableCell[];
+}
+
+/**
+ * Represents a single cell in a portable text table.
+ */
+export interface PortableTextTableCell extends ArbitraryTypedObject {
+  _type: "cell";
+  /**
+   * Number of blocks that belong to a table cell.
+   * Helps with table resolution.
+   */
+  childBlocksCount: number;
+  /**
+   * All blocks belonging to a cell.
+   */
+  content: PortableTextBlock[];
+}
+
+/**
+ * Represents a component or a linked item used in rich text.
+ */
+export interface PortableTextComponent extends ArbitraryTypedObject {
+  _type: "component";
+  /**
+   * Reference to a component or a linked item.
+   */
+  component: Reference;
+}
+
+/**
+ * Represents a mark that styles the text (`strong`, `em`, etc.)
+ */
+export interface PortableTextStyleMark extends ArbitraryTypedObject {
+  _type: "mark";
+  value: "strong" | "em" | "sub" | "sup" | "code"; // TODO: formerly TextStyleElement, is circular import okay in this case?
+}
+
+/**
+ * Represents a mark that references a link type object in `markDefs` array via a guid.
+ */
+export interface PortableTextLinkMark extends ArbitraryTypedObject {
+  _type: "linkMark";
+  /**
+   * Short guid pointing to a corresponding object in `markDefs` array.
+   */
+  value: string;
+  /**
+   * Number of spans the link extends to.
+   *
+   * This is required for merging process as one link can potentially have multiple spans,
+   * in case parts of the link text are styled.
+   */
+  childCount: number;
+}
+
+/**
+ * Represents a block, usually a paragraph or heading.
+ *
+ * Narrows the `_type` to `block` for type guard purposes.
+ */
+export interface PortableTextStrictBlock
+  extends Omit<PortableTextBlock, "_type">, ArbitraryTypedObject {
+  _type: "block";
+}
+
+/**
+ * Represents a list item block. Similar to regular block but requires `listItem` property.
+ *
+ * Narrows the `_type` to `block` for type guard purposes.
+ */
+export interface PortableTextStrictListItemBlock
+  extends Omit<PortableTextListItemBlock, "_type">, ArbitraryTypedObject {
+  _type: "block";
+}
+
+export type PortableTextLink =
+  | PortableTextInternalLink
+  | PortableTextExternalLink;
+
+export type PortableTextMark = PortableTextLinkMark | PortableTextStyleMark;
+
+/**
+ * Union of all default portable text object types.
+ */
+export type PortableTextObject =
+  | PortableTextLink
+  | PortableTextComponent
+  | PortableTextImage
+  | PortableTextTable
+  | PortableTextTableRow
+  | PortableTextTableCell
+  | Reference
```

Comment:
Since the core function now returns the `PortableTextObject`, the type should only list the blocks that can be returned from the function so that it is accurate. I believe that `Reference` and the rows and cells cannot be in the array returned from the function, right? It would better to split the type and use the other types only where they really are.

Reply:
split into PortableTextObject and PortableTextInternalObject, both unionized under PortableTextItem